### PR TITLE
Migrate tests-legacy directory to tests directory

### LINKED ISSUE #23462: Migrate tests-legacy directory to tests directory
- [x] Migrate create-test-db command
  - #25843
- [x] Migrate `phpunit-legacy` (Tests: 948, Assertions: 3219) => :wave:
  - #24095
  - #24121
  - #24188
  - #25127
  - #25238
- [x] Migrate `phpunit-admin` (Tests: 155, Assertions: 275) => :wave:
  - #25238
- [x] Migrate `phpunit-sf` (Tests: 25, Assertions: 104) => :wave: 
  - #25127
- [x] :x: Migrate `phpunit-endpoints` (Tests: 16, Assertions: 31)
- [x] Migrate `phpunit-controllers` (Tests: 28, Assertions: 28) => :wave: 
    - #25196
- [x] Clean modules ressources
- [x] Remove tests with gamification
    - #25721
- [x] Update docs
    - PrestaShop/docs#1244
- [x] Move LegacyContext in Integrations Tests directory
    - #26533

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 8.x
| Description?  | Migrate tests-legacy directory to tests directory
| Fixed ticket? | Fixes PrestaShop/PrestaShop#23462

### CODE REVIEW

File: testing/introduction.md [hunk 1]
```diff
@@ -58,16 +58,14 @@ UI tests rely on [Playwright][4].
...
-
-Tests that we are not satisfied by remain in the `tests-legacy` folder and will be, one by one, replaced by tests of a higher quality inside the `tests` folder.
-
-We keep the legacy tests as they have value (they can detect mistakes in the modifications we bring to the codebase) but they are not being updated anymore as we want to replace them.
-
-#### What was wrong with these legacy tests?
-
-Theses tests were powered by PHPUnit but were not unit tests: they were integration tests, which means they would test several classes together and would run additional services such as a database. This has made this test quite complex and sometimes lead to random failures.
-
+Actually, `tests-legacy` contains only tests targetting these endpoints:
```
Comment (matks) [inline testing/introduction.md]:
```suggestion
Folder `tests-legacy` contains only tests targeting these endpoints: 
```

**Review by matks — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by eternoendless — CHANGES_REQUESTED** [review]
Watch out, there's a new submodule that you need to remove in `src/themes/ps-docs-theme`

**Comment by Progi1984** [comment]
@eternoendless It's fixed. Thanks for the eagle :eye:

File: testing/introduction.md (hunk 1 above)
Comment (eternoendless) [reply testing/introduction.md]: "Actually" means "in fact". I suppose you meant "currently"
```suggestion
Currently, the `tests-legacy` folder only contains tests targeting these endpoints: 
```

File: testing/how-to-launch-tests.md [hunk 2]
```diff
@@ -10,17 +10,17 @@ weight: 10
...
 At least four test suites are available, testing different parts of PrestaShop:
 
-* `Legacy tests`: heritage from PrestaShop 1.6: mix of unit, integration and functional tests;
-* `Legacy controllers`: added to help with the Symfony migration, ensures that old Back Office controllers are still runnable;
-* `Admin tests`: tests `src/Core` and `src/Adapter` classes;
-* `Symfony specific tests`: test classes from `src/PrestaShopBundle`
+* `Unit tests`
+* `Integrations tests`
+* `Functional tests`
+* `Legacy endpoints`
```
Comment (eternoendless) [inline testing/how-to-launch-tests.md]: Why are these highlighted as code? 🤔

File: testing/introduction.md (hunk 1 above)
Comment (Progi1984) [reply testing/introduction.md]: Fixed

File: testing/how-to-launch-tests.md (hunk 2 above)
Comment (Progi1984) [reply testing/how-to-launch-tests.md]: Fixed

**Review by eternoendless — APPROVED** [review]
LGTM!

**Comment by eternoendless** [comment]
Thank you @Progi1984

**Pull request merged**